### PR TITLE
Widen ascii to utf16

### PR DESCRIPTION
**Pull request by pgovind**
Ready for review now. 

Implements WidenAsciiToUtf16 from #35034

Perf:
```
summary:
better: 1, geomean: 2.172
total diff: 1

No Slower results for the provided threshold = 0.01% and noise filter = 0.3ns.

| Faster                                                       | base/diff | Base Median (ns) | Diff Median (ns) | Modality|
| ------------------------------------------------------------ | ---------:| ----------------:| ----------------:| --------:|
| System.Text.Experimental.Perf.IsNormalized_WidenAsciiToUtf16 |      2.17 |         65021.15 |         29935.14 |         |

No file given
```

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs [hunk 1]
```diff
@@ -1597,7 +1597,18 @@ public static unsafe nuint WidenAsciiToUtf16(byte* pAsciiBuffer, char* pUtf16Buf
             goto Finish;
         }
 
-        private static unsafe nuint WidenAsciiToUtf16_Sse2(byte* pAsciiBuffer, char* pUtf16Buffer, nuint elementCount)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool ContainsNonAsciiValue(Vector128<byte> value)
```
Comment (eiriktsarpalis) [inline src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs:1601]: Nit: rename to `ContainsNonAsciiByte`?

**Review by eiriktsarpalis — APPROVED** [review]

**Comment by kunalspathak** [comment]
Same here. Please add the benchmark in `dotnet/performance`.

**Review by echesakov — APPROVED** [review]
LGTM (with nits)

**Comment by pgovind** [comment]
Thanks for the reviews @eiriktsarpalis @echesakovMSFT and @kunalspathak

**Comment by pgovind** [comment]
CI failures are unrelated. Merging

**Comment by kunalspathak** [comment]
Could you double check the benchmark `IsNormalized_WidenAsciiToUtf16()` that you added in https://github.com/dotnet/performance/pull/1445 to see if it is actually hitting the code path. You don't do anything with the return value so does JIT optimize away calls that would have otherwise called the code path you wanted to measure?

**Comment by pgovind** [comment]
Yup, it's hitting the code path. I added an exception that triggered by the benchmarking code:

```
// BeforeAnythingElse


Unhandled exception.
   Cannot print exception string because Exception.ToString() failed.
ExitCode != 0
// Benchmark Process 4827 has exited with code 134
No more Benchmark runs will be launched as NO measurements were obtained from the previous run!
```

**Pull request merged**